Dockerfile

```diff
@@ -1,15 +1,25 @@
-FROM alpine:edge
+FROM node:7.10-alpine
+LABEL maintainer "Tim Evans <tim.c.evans@me.com>"
 
-WORKDIR /usr/src/app
+WORKDIR /usr/src/app/
 
-RUN apk --update add nodejs git ruby ruby-dev ruby-json build-base && \
-  apk del build-base && rm -fr /usr/share/ri
+COPY engine.json ./
+COPY package.json yarn.lock ./
 
-RUN npm install -g mrb/ember-watson.git#7dacf719
-
-COPY . /usr/src/app
+RUN apk --update add bash curl git gnupg jq tar && \
+    curl -o- --location --silent https://yarnpkg.com/install.sh | /bin/bash --noprofile && \
+    yarn install --production --frozen-lockfile --no-progress && \
+    yarn cache clean && \
+    version="v$(npm -j ls ember-watson | jq -r '.dependencies["ember-watson"].version')" && \
+    cat ./engine.json | jq ".version = \"$version\"" > /engine.json && \
+    apk del bash curl git gnupg jq tar
 
 RUN adduser -u 9000 -D app
+COPY . ./
+
 USER app
 
-CMD ["/usr/src/app/bin/codeclimate-watson"]
+WORKDIR /code
+VOLUME /code
+
+CMD ["/usr/src/app/bin/ember-watson.js"]
```

Comment:
What do you think of keeping the executable `/usr/src/app/bin/codeclimate-watson`, as we're running the engine wrapper script (not ember-watson itself) and we can keep the executable name language-agnostic.